### PR TITLE
[Merged by Bors] - chore(Algebra/Polynomial/RingDivision): move lemmas earlier

### PR DESCRIPTION
**Pull request by YaelDillies**
`Algebra.Polynomial.RingDivision` is a bag grab of lemmas with absolutely no running theme. This PR moves two thirds of them to earlier files with no proof change.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mattrobball** [comment]
Thanks!

Please merge after CI succeeds. 

bors delegate+

File: Mathlib/Algebra/Polynomial/RingDivision.lean [hunk 1]
```diff
@@ -63,26 +42,6 @@ theorem degree_pos_of_aeval_root [Algebra R S] {p : R[X]} (hp : p ≠ 0) {z : S}
...
-    · exact
-      (div_modByMonic_unique (p₁ /ₘ q + p₂ /ₘ q) _ hq
-          ⟨by
-            rw [mul_add, add_left_comm, add_assoc, modByMonic_add_div _ hq, ← add_assoc,
-              add_comm (q * _), modByMonic_add_div _ hq],
-            (degree_add_le _ _).trans_lt
-              (max_lt (degree_modByMonic_lt _ hq) (degree_modByMonic_lt _ hq))⟩).2
-  · simp_rw [modByMonic_eq_of_not_monic _ hq]
-
 theorem smul_modByMonic (c : R) (p : R[X]) : c • p %ₘ q = c • (p %ₘ q) := by
```
Comment (b-mehta) [inline Mathlib/Algebra/Polynomial/RingDivision.lean:45]: It looks a little weird to me to move add_ neg_ and sub_ but not smul_ and the bundled version. What's stopping us from moving these two as well?

File: Mathlib/Algebra/Polynomial/RingDivision.lean [hunk 2]
```diff
@@ -455,35 +104,10 @@ theorem rootMultiplicity_eq_rootMultiplicity {p : R[X]} {t : R} :
...
-  by_cases h : p = 0
-  · simp only [h, rootMultiplicity_zero, natTrailingDegree_zero]
-  refine le_antisymm ?_ ?_
-  · rw [rootMultiplicity_le_iff h, map_zero, sub_zero, X_pow_dvd_iff, not_forall]
-    exact ⟨p.natTrailingDegree,
-      fun h' ↦ trailingCoeff_nonzero_iff_nonzero.2 h <| h' <| Nat.lt.base _⟩
-  · rw [le_rootMultiplicity_iff h, map_zero, sub_zero, X_pow_dvd_iff]
-    exact fun _ ↦ coeff_eq_zero_of_lt_natTrailingDegree
-
 theorem rootMultiplicity_eq_natTrailingDegree {p : R[X]} {t : R} :
```
Comment (b-mehta) [inline Mathlib/Algebra/Polynomial/RingDivision.lean:107]: This one and the previous clearly have strong commonalities, if we can't keep them together then let's be sure to add docstrings referencing the other

File: Mathlib/Algebra/Polynomial/RingDivision.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/Polynomial/RingDivision.lean:45]: Requires `Algebra.Polynomial.AlgebraMap` on top

**Comment by YaelDillies** [comment]
> Please make sure all authors on the deleted file are accounted for and then merge after CI succeeds.

Not sure what "accounted for" means here but
1. The file is not deleted
2. The copyright header (and module doc!) is manifestly a copy-paste of `Algebra.Polynomial.Roots`
3. From a quick read through the file's history, most of it was written after the first commit, so the author list has no chance of being accurate
4. No series of lemmas that was moved makes up for more than a small fraction of their new file, so changing the list of authors there wouldn't be appropriate
5. I am lazy and not looking forward to figuring out authorship of a hundred lemmas 😅

In short: I won't do anything about authors.

**Comment by YaelDillies** [comment]
bors merge
